src/symbolic_choice.ml

```diff
@@ -144,6 +144,55 @@ end = struct
 end
 [@@inline]
 
+module Minimalist = struct
+  type err =
+    | Assert_fail
+    | Trap of Trap.t
+
+  type 'a res = ('a, err) Stdlib.Result.t
+
+  type 'a t = M of (Thread.t -> 'a res * Thread.t) [@@unboxed]
+
+  let return v = M (fun t -> (Ok v, t))
+
+  let run_minimalist (M v) st = v st
+
+  let bind v f =
+    M
+      (fun init_s ->
+        let v_final, tmp_st = run_minimalist v init_s in
+        match v_final with
+        | Ok v_final -> run_minimalist (f v_final) tmp_st
+        | Error (Trap t) -> (Error (Trap t), tmp_st)
+        | Error Assert_fail -> (Error Assert_fail, tmp_st) )
```

Comment:
Please, try to remember using the `as` pattern to avoid extra-allocations :) :

```ocaml
match v_final with
| Ok v_final -> run_minimalist (f v_final) tmp_st
| Error (Trap _t) as e -> (e, tmp_st)
| Error Assert_fail as e -> (e, tmp_st) )
```

Moreover, the last two cases can be merged:

```ocaml
match v_final with
| Ok v_final -> run_minimalist (f v_final) tmp_st
| Error _ as e -> (e, tmp_st)
```